### ISSUE TITLE
Can I get this for a kobo ereader?

### ISSUE THREAD

**Issue by pokemoneuro** [wjdp/gotdict#40]
Yo, I was wondering how I can use this as a dictionary on a Kobo device.

**Comment by wjdp** [comment]
We currently only have builds for Amazon's Kindle (mobi format) as that's the only eReader I own. I'd love to add support for other devices and from a cursory search Kobo uses EPUB which should also give us support for quite a few other devices as well.

Dictionary formatting tends to be more complex than building a standard ebook so we'd need someone both with one of these devices and the time to get a build up and running. @pokemoneuro would you be interested in taking this on?

**Comment by pokemoneuro** [comment]
I'm not very experienced in this field and I don't really know how everything sticks together, sorry. I'm only 16, still learning these things. Also, I don't own my own e-reader and when I'm gonna buy one it'll probably be a Kindle anyway xD. Thanks anyway for informing me!

**Comment by wjdp** [comment]
No trouble. Half the point of these projects is to play around with some tech without any real consequence.

> I'm not very experienced in this field

Neither was anyone else when they started :wink: This is my first and only eBook...thing.

Anyway find something you're interested in and have a fiddle!

**Comment by pokemoneuro** [comment]
True, of course, you gotta start somewhere. I'm going on vacation tomorrow so it doesn't seem like the right time. I also don't have a device to test it on, only my grandpa's which I'm using for on the vacation. Anyway, if I ever have a device to experiment with I can look, but like I said, I'll probably buy a Kindle when I buy an eReader for myself.

**Comment by geek1011** [comment]
I've made a [Kobo version](https://cloud.drone.io/geek1011/dictutil/90/6/6) of GOTDict as part of my work on [dictutil](https://github.com/geek1011/dictutil).

**Comment by wjdp** [comment]
@geek1011 Cool! Would be happy to accept a PR that adds this build into CI/CD.

**Comment by geek1011** [comment]
I currently build it as part of dictutil, and want to keep it that way for now (dictutil is still in progress, and I haven't made a decision about images yet: see geek1011/dictutil#1). After I release the first stable version, I'll submit a PR to add it to the CI here.

I'll also submit a PR to fix a few bugs in gotdict which I found while working on gotdict-convert (which parses it quite strictly). See the `FIX:` comments in [gotdict.go](https://github.com/geek1011/dictutil/blob/master/examples/gotdict-convert/gotdict.go).

**Comment by geek1011** [comment]
Images have been fixed in firmware 4.20.14601. I'll make another release which adds them again (I'll build it with and without images).

**Comment by geek1011** [comment]
I have released dictutil v0.1.0 stable. It's pretty much finished now.

**Comment by wjdp** [comment]
Hey @pgaskin I spotted you've got a build working on https://pgaskin.net/dictutil/examples/gotdict-convert.html

I've just moved the CI from Travis to GitHub actions. Do you want to have a look at integrating the Kobo version here?

**Comment by pgaskin** [comment]
> Do you want to have a look at integrating the Kobo version here?

Yep, I'll take a look when I work on v3 dictionary support in dictutil.
